src/codeu/chat/server/View.java

```diff
@@ -78,6 +78,140 @@ public View(Model model) {
   @Override
   public Message findMessage(Uuid id) { return model.messageById().first(id); }
 
+  @Override
+  public Collection<String> userStatusUpdate(String name, Uuid owner) {
+
+    // Given a specified user's name and the UUID of the user that wants to
+    // request a status update on them, return a collection of conversation
+    // titles of conversations that the specified user has either added messages
+    // to or created since the last time the user requesting a status update has
+    // requested a status update on the specified user.
+
+    Collection<ConversationPayload> allConversations = all(model.conversationPayloadById());
+    Collection<String> contributions = new ArrayList<String>();
+
+    final User foundOwner = model.userById().first(owner);
+    final User foundUser = model.userByText().first(name);
+
+    // check if the specified user can be found in Model
+    if(foundUser != null) {
+      // check if the specified user is in the current user's user interests
+      if(foundOwner.UserSet.contains(foundUser.id)) {
+        // the last time that the current user requested a status update for the specified user
+        final Time lastUpdate = foundOwner.UserUpdateMap.get(foundUser.id);
+        // go through all of the conversations stored in Model
+        contributions = searchContributions(lastUpdate, foundUser.id);
+        // if after going through everything and no contributions are found, add the note to the collection
+        if(contributions.isEmpty()) {
+          contributions.add("(No recent conversations)");
+        }
+        // finally, update the time that status update was last requested for the specified user to now
+        foundOwner.UserUpdateMap.put(foundUser.id, Time.now());
```

Comment:
UPDATE AGAIN: Disregard this, error was in Chat. Accidentally called the method twice in a row.